sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -396,23 +440,50 @@ private[spark] object HiveUtils extends Logging {
         config = configurations,
         barrierPrefixes = hiveMetastoreBarrierPrefixes,
         sharedPrefixes = hiveMetastoreSharedPrefixes)
+    } else if (hiveMetastoreJars == "path") {
+      // Convert to files and expand any directories.
+      val jars =
+        HiveUtils.hiveMetastoreJarsPath(sqlConf)
+          .flatMap {
+            case path if Utils.isWindows =>
+              addLocalHiveJars(new File(path))
+            case path =>
+              val uri = new Path(path).toUri
+              uri.getScheme match {
```

Comment:
> I don't understand why we need to check the scheme and do things differently. `spark.read.parquet(...)` can support any schema and internally just use the Hadoop DFS API. Can't we do that as well here?
> 
> Can you point to other places in Spark that do similar things to support this PR?

In datasource file index, path is fully qualified URL to indicate other file systems.
So it support  `file://  | local:// `, but without schema, it will be handled as hdfs path.
One way to avoid check schema,  as https://github.com/apache/spark/pull/29881#discussion_r496649487 mentioned, we host user use fully qualified URL.

Reply:
how about current change? Make the logic more simpler.